x-pack/platform/plugins/shared/lens/public/trigger_actions/open_in_discover_action.ts

```diff
@@ -53,15 +54,9 @@ export const createOpenInDiscoverAction = (
         throw new IncompatibleActionError();
       return hasDiscoverAccess && Boolean((embeddable as LensApi).canViewUnderlyingData$);
     },
-    subscribeToCompatibilityChanges: (
-      { embeddable }: EmbeddableApiContext,
-      onChange: (isCompatible: boolean, action: Action<EmbeddableApiContext>) => void
-    ) => {
-      if (!typeof (embeddable as LensApi).canViewUnderlyingData$)
-        throw new IncompatibleActionError();
-      return (embeddable as LensApi).canViewUnderlyingData$.subscribe((canViewUnderlyingData) => {
-        onChange(canViewUnderlyingData, actionDefinition);
-      });
+    getCompatibilityChangesSubject: ({ embeddable }: EmbeddableApiContext) => {
+      if (!typeof (embeddable as LensApi).canViewUnderlyingData$) return;
+      return (embeddable as LensApi).canViewUnderlyingData$.pipe(map(() => undefined));
```

Comment:
What's the purpose of returning `undefined` instead of `boolean`?

Reply:
`subscribeToCompatibilityChanges` does 2 things
1) create subject that notifies consumers of state changes that potentially could change compatibility
2) subscribes to subject and re-calculates compatibility. This is where the problem lies, introducing a second location for compatibility logic.

`getCompatibilityChangesSubject`, that only does `1)` above. Then consumers of the subject will be responsable for `2)` and can re-calculate compatibility by calling `action.isCompatible`.

Reply:
Oh ok nice, I appreciate the explanation!